src/librawspeed/decoders/DngDecoderSlices.cpp

```diff
@@ -418,10 +419,9 @@ void DngDecoderSlices::decodeSlice(DngDecoderThread* t) {
     while (!t->slices.empty()) {
       DngSliceElement e = t->slices.front();
       t->slices.pop();
-      LJpegPlain l(*mFile, e.byteOffset, e.byteCount, mRaw);
-      l.mDNGCompatible = mFixLjpeg;
+      DngDecompressor d(*mFile, e.byteOffset, e.byteCount, mRaw);
```

Comment:
`LJpegDecompressor` is the base class of 3 classes:
  * Cr2Decompressor
  * DngDecompressor
  * HasselbladDecompressor

They are _all_ implementing a form of 'lossless jpeg comression', that is why they share a common base class, but they differ in details specific to the different formats. Hence I think the naming makes absolute sense. You could argue that DNG also has other compression options and calling this one `DngDecompressor` is misleading in the way that it only deals with one variant. So maybe calling it `DngLJpegDecompressor` would be more to your liking?

Introducing the term 'Abstract' would certainly be possible but it is also not used in other places, like `RawDecoder`. Maybe you want to generally introduce this kind naming hint across the library. In that case I'd consider two alternatives, though: either a postfix `Base` or a prefix `I` for interface. Either way: please do that later as a separate pr. To quote yourself: 'one step at a time'.